libsignify/src/encoding.rs

```diff
@@ -215,20 +215,18 @@ fn read_base64_contents(encoded: &str) -> Result<(Vec<u8>, u64), Error> {
 
     let base64_line = lines.next().ok_or(FormatError::MissingNewline)?;
 
-    if base64_line.is_empty() {
-        return Err(FormatError::LineLength.into());
-    }
-
     let data = base64::decode(base64_line.trim_end()).map_err(|_| FormatError::Base64)?;
 
     match data.get(0..2) {
         // Make sure the specified algorithm matches what we support
         Some(alg) if alg == PKGALG => {
-            // Can't panic, we know there are two lines present.
+            // Find the last newline in the standard encoding scheme. Any bytes after it would be an embedded file.
             let mut newlines = encoded.chars().enumerate().filter(|(_, val)| *val == '\n');
-            let remaining = newlines.nth(1).unwrap().0 as u64;
+            let remaining = newlines.nth(1).ok_or(FormatError::MissingNewline)?.0;
 
-            Ok((data, remaining + 1))
+            // The position of the last newline is `remaining`, so subtract that from the total length.
+            // +1 to account for the trailing newline that is included in the positioning logic.
+            Ok((data, (encoded.len() - (remaining + 1)) as u64))
```

Comment:
I lost track of what we even want to return here, which means this should be assigned to a variable with a good name to express what it does.

Reply:
Yeah, pretty much :/ Going back and looking at it two days later I couldn't really figure out what I was going for and the comment was garbage.

My structure encoding/decoding tests sent me for quite a trip here lmao. They were failing otherwise (`full-cycle` and them traded off passing with each change) and it turns out the assertion there was just plain wrong.

The new simple code makes this clear imo, but now its just equivalent to what `std::io::Read` would return and the docs are updated to reflect that. Also updated the CLI to make it wayyy clearer. Looking at `read_base64_file` in the CLI made the revealed my mistake in an instant.